unified-runtime/source/adapters/level_zero/v2/command_list_manager.hpp

```diff
@@ -36,8 +36,7 @@ struct ur_command_list_manager {
   ur_command_list_manager(ur_context_handle_t context,
                           ur_device_handle_t device,
                           v2::raii::command_list_unique_handle &&commandList,
-                          v2::event_flags_t flags = v2::EVENT_FLAGS_COUNTER,
-                          ur_queue_t_ *queue = nullptr);
+                          v2::event_flags_t flags, ur_queue_t_ *queue);
```

Comment:
We should try to remove queue from the command list manager and events altogether.

Instead, recordStartTimestamp should take the queue as a parameter, and store it in the event_profiling_data_t (to make sure the `queue` variable is never used for anything else).

Reply:
Unfortunately, we also need queue for getInfo: https://github.com/intel/llvm/blob/sycl/unified-runtime/source/adapters/level_zero/v2/event.cpp#L265 and in getProfilingInfo, when recordStartTimestamp was NOT called (we rely on queue for profiling info).

I could create a wrapper on event_pool (queue_event_pool) that accepts a queue in it's ctor and then queue_event_pool::allocate() creates an event with proper queue already set (so that, command_list_manager does not need to hold any queue references) but that's still circular reference, only hidden in one extra layer.

Reply:
:( 
> I could create a wrapper on event_pool (queue_event_pool) that accepts a queue in it's ctor and then queue_event_pool::allocate() creates an event with proper queue already set (so that, command_list_manager does not need to hold any queue references) but that's still circular reference, only hidden in one extra layer.

Nah, let's put a pin on this for now. But in general, I like the idea of specializing the event pool to contain the queue (or not... like in the case of command buffers). I'm resigned to the fact that events will need circular references to queues, but the mutual references between queue and cmdlist manager feel especially iffy.

 I think we might want to change the queue in event to be an optional, to indicate that it may not always exist.